---
title: "Maize Selection Tests"
output: pdf_document
---

Investigating genome size variation in maize and teosinte.  Ultimately, we want to identify whether our nuclear phenotypes (genome size, repeat abundance) are changing more rapidly than we would expect given kinship matrices.  We have data from approximately 90 Zea landraces with point estimates for phenotype+genotype, and <10 populations each of parv and mex, with population averages for each.  First, I plot the data to visualize the variation across altitudinal clines.  We want to get a sense of the altitudinal trends.

Read in the maize data with this chunk.

```{r}
setwd("~/Documents/Projects/Genome_Size_Analysis")
library(ggplot2)
#install.packages(gridextra)
library(gridExtra)
dataall <- read.csv("Master_Data_noNA.csv")
dataal <- subset(dataall, dataall$X1C_GS!="NA")
data <- subset(dataal, dataal$X1C_GS<3.6)
dmays <- subset(data, data$Species=="mays")
dmayst <- subset(dmays, dmays$Region!="SWUS")
dmex <- subset(data, data$Species=="mexicana")
dparv <- subset(data, data$Species=="parviglumis")
dteo <- rbind(dparv,dmex)
```

Plot the variation.  Here, we note that several repeats have opposing clines.  Furthermore, we also find that the teosintes have a different trend in terms of within species genome size cline.  Though they do show that the highland mex is of average smaller size than the lowland parv, both mex and parv increase in genome size across altitude. 

```{r}
p1 <- ggplot(dmayst, aes(Region, X180knobMB, color=Region)) + geom_boxplot()+ ylab("180 Knob MB")+ theme(legend.position="none")
p <- ggplot(dmayst, aes(Region, TotallTeMB, color=Region)) + geom_boxplot()+ ylab("Total TE MB")+ theme(legend.position="none")
grid.arrange(p1,p, ncol=2)


p1 <- ggplot(dteo, aes(Region, X1C_GS, color=Region)) + geom_boxplot()+ ylab("1C GS")
p1
p2 <- ggplot(dteo, aes(Altitude, X1C_GS, color=Species)) + geom_point()+ ylab("1C GS") + theme(legend.title=element_blank()) + geom_smooth(method=lm)
p2
p <- ggplot(dteo, aes(Altitude, TR1MB, color=Species)) + geom_point()+ ylab("MB TR1") + theme(legend.title=element_blank()) + geom_smooth(method=lm)
p
p2 <- ggplot(dteo, aes(Altitude, X180knobMB, color=Species)) + geom_point()+ ylab("MB 180Knob") + theme(legend.position="none") + geom_smooth(method=lm)
p2
p <- ggplot(dteo, aes(Altitude, TR1MB, color=Species)) + geom_point()+ ylab("MB TR1") + theme(legend.position="none") + geom_smooth(method=lm)
p
grid.arrange(p2,p, ncol=2)
p <- ggplot(data, aes(Altitude, TotallTeMB, color=Species)) + geom_point()+ ylab("MB TE") + theme(legend.title=element_blank()) + geom_smooth(method=lm)
p
```

Feel free to alter pieces of the plotting code to see trends in different repeats.  Given the observed trends, we can now advance to studying selection on genomic elements.  Read in the genotype data with this code, and generate the genetic matrix with the following code.

Of importance is that we have point estimates for landraces and population averages for the teosintes.  RRBLUP will take in allele frequencies and generate kinship matrices, so we feed it the point estimates in maize, and then the population averages in teosinte.  NOTE: GENERATE POP AVERAGES FOR TEOSINTE OVER XMAS.

#go back to teosinte data, convert to frequencies of allele * 2 so that we get the # of alleles in each individual

```{r}
setwd("~/Documents/Projects/Genome_Size_Analysis/Github_ParallelGS/SelectionTests/")
library("rrBLUP")
geno <- read.csv("~/Documents/Projects/Genome_Size_Analysis/Github_ParallelGS/SNP_data/Landrace_noSWUS_matrix.csv",header=TRUE,row.names=1)
dt <-t(geno)
A <- A.mat(dt)
```

Read in the phenotype data, and make sure the order of the samples are the same as order in the genetic matrix.  To do so, we use the tmp variables that store the proper order sample IDs in the maize samples.  Note, this is just for the maize samples, the teosintes are done later.

```{r}
pheno <-read.csv("~/Documents/Projects/Genome_Size_Analysis/Github_ParallelGS/PhenotypeData/Landraces_noSWUS_pheno.csv",header=TRUE)
#to make sure order is the same
tmp1 <- as.data.frame(colnames(geno))
names(tmp1)[1] <- "names"
tmp2 <- as.data.frame(pheno$FullID)
tmp <- setdiff(tmp1,tmp2)

phenoorder <- merge(tmp1,pheno, by.x="names", by.y="FullID",sort=FALSE)
```

This next chunk was our first iteration of Jeremy's code, we can ignore it for now and step to the current version.  Kept here for archaelogy.

```{r}
#library ( mvtnorm )
#EnvVarTest <- function ( phenos , kinship.mat , test.vector, verbose=F ) {
  
  # 'phenos' is a vector containing the phenotype 
  # (i.e. number of repeats) for each individual; dimensions are N x 1
  # 'kinship.mat' is the kinship matrix; dimensions are N x N;
  # rows and columns need to be in the same order as the phenotypes in the vector
  # test.vector is the environmental factor of interest (in this case altitude)
  
#   eigs <- eigen ( kinship.mat ) 
#   # get eigendecomposition of kinship matrix
#   rt.inv <- eigs$vec %*% diag ( 1/sqrt(eigs$val) ) %*% t ( eigs$vec ) # standardizing distances across each of the eigenvectors so the scale is the same. #change 1 and change 2 dealing with the rotation in and out of transformation 
#   # calculate inverse of the square root matrix
#   rotated.phenos <- t ( rt.inv ) %*% phenos 
#   # rotate phenotypes from population space into principal component space
#   test.vector <- test.vector / (sqrt ( 2 * sum ( test.vector^2 ) ) ) 
#   # scale to be unit length after rotation
#   #recover()
#   rotated.vector <- rt.inv %*% test.vector 
#   # rotate environmental variable from population space into principal component space
#   model <- lm ( rotated.phenos ~ 1+rotated.vector) 
#   # fit regression model
#   r.sq <- cor.test ( rotated.phenos , rotated.vector )$estimate^2 
#   # get r^2
#   ANOVA <- anova ( model ) 
#   # get p value
#   if(verbose){ print(ANOVA) }
#   return ( c ( model$coef[2] , r.sq , ANOVA[5][[1]][1] )) # return 
# #}
```

EnvVarTest, the code written by Jeremy Berg to run an Fst-Qst like test on our data.  Comments are included within the code to explain what it is doing at each step.

```{r}
library ( mvtnorm )


EnvVarTest <- function ( phenos , kinship.mat , test.vector, verbose=F ) {
  
	# 'phenos' is a vector containing the phenotype (i.e. number of repeats) for each individual; dimensions are N x 1
	# 'kinship.mat' is the kinship matrix; dimensions are N x N; rows and columns need to be in the same order as the phenotypes in the vector
	# test.vector is the environmental factor of interest (in this case altitude)

	#recover()
	eigs <- eigen ( kinship.mat ) ## get eigendecomposition of kinship matrix
	rt.inv <- eigs$vec %*% diag ( 1/sqrt(eigs$val) ) %*% t ( eigs$vec )# calculate inverse of the square root matrix
	
  cent.test.vector <- test.vector - mean ( test.vector )
  cent.phenos <- phenos - mean ( phenos )
  
  rotated.phenos <- rt.inv %*% cent.phenos # rotate phenotypes from population space into principal component space
	unit.test.vector <- cent.test.vector / (sqrt ( 2 * sum ( cent.test.vector^2 ) ) ) # scale to be unit length after rotation
	#recover()
	
  
  
  
  rotated.vector <- rt.inv %*% cent.test.vector # rotate environmental variable from population space into principal component space
	model <- lm ( rotated.phenos ~ rotated.vector) # fit regression model
	r.sq <- cor.test ( rotated.phenos , rotated.vector )$estimate^2 # get r^2
	ANOVA <- anova ( model ) # get p value
	
	# expected fraction of variance associated with environmental variable under neutrality
	F.env <- ( t ( unit.test.vector ) %*% kinship.mat %*% unit.test.vector ) / sum ( diag ( kinship.mat ) )
	sums.sq <- cumsum ( ANOVA[2] ) 
	
	# fraction of variance associated with environmental variable for trait
	Q.env <- sums.sq[1,]/ ( sums.sq[2,] )
if(verbose==T){print(ANOVA)}
	return ( c ( model$coef[2] , r.sq , ANOVA[5][[1]][1] , ANOVA[3][2,] , F.env , Q.env) ) # return regression coefficient, r-squared, p value, estimate of Va, F statistic, Q statistic

}
```

Regression coefficient is the B

Now run the equation on each of the nuclear phenotypes.  Altitude is our environmental variable, and we will expand on doing this with other bioclim variables.  The MB signifies megabases of the repeat, and the . is a replacement for the % symbol.

```{r}
EnvVarTest(phenoorder$X180knobMB,A,phenoorder$Altitude)
EnvVarTest(phenoorder$X180knob.,A,phenoorder$Altitude)
EnvVarTest(phenoorder$TotallTeMB,A,phenoorder$Altitude)
EnvVarTest(phenoorder$TotallTe.,A,phenoorder$Altitude)
EnvVarTest(phenoorder$TR1MB,A,phenoorder$Altitude) 
EnvVarTest(phenoorder$TR1.,A,phenoorder$Altitude) 
EnvVarTest(phenoorder$CentCMB,A,phenoorder$Altitude)
EnvVarTest(phenoorder$CentC.,A,phenoorder$Altitude)
EnvVarTest(phenoorder$CRMallMB,A,phenoorder$Altitude)
EnvVarTest(phenoorder$CRMall.,A,phenoorder$Altitude)
EnvVarTest(phenoorder$X1C_GS,A,phenoorder$Altitude)
<<<<<<< HEAD


#plot(phenoorder$TotallTe. ~ phenoorder$Altitude)
#lm(phenoorder$TotallTe. ~ phenoorder$Altitude)

=======


#plot(phenoorder$TotallTe. ~ phenoorder$Altitude)
#lm(phenoorder$TotallTe. ~ phenoorder$Altitude)

>>>>>>> 734b6aa2fbd6492078fbcbeff73f551250c1a4db
```

Jeremy's simulation code, need to acknowledge NIH grant, reminder for paper.

```{r}
# load
# 
# my.betas <- c ( seq ( 0 , 1 , by = 0.1 ), seq ( 1.2 , 2, 0.2 ) )
# env <-rnorm ( n = 83 , 0 , sd = 1 ) # different individuals
# tmp <- list ()
# power <- numeric ( length ( my.betas ) )
# for ( i in 1 : length ( my.betas ) ) {
# 
#   beta <- my.betas [ i ]
# 	mu <- rep ( 0 , 152 )
# 	my.mean <- mu + beta * env
# 	test.data <- rmvnorm ( n = 1000 , mean = my.mean , sigma = this.cov.mat )
# 	
# 	tmp [[ i ]] <- matrix ( NA , nrow = 1000 , ncol = 6 )
# 	for ( j in 1 : 1000 ) {
# 		tmp [[ i ]] [ j , ] <- EnvVarTest ( test.data[ j , ] , this.cov.mat , env ) 
# 	}
# 	#hist ( tmp [[ i ]] [ , 3 ] , breaks = 50 )
# 	
# 	power [ i ] <- sum ( tmp [[ i ]] [ , 3  ] < 0.05 ) / 1000
# 
# }
# 
# Q.stat <- do.call ( cbind , lapply ( tmp , function ( x ) x [ , 6 ] ) )
# 
# par ( mfrow = c ( 2, 1 ) )
# plot ( my.betas , power , xlab = expression ( beta ) , ylab = "Power"  , type = "l" , lwd = 2 )
# 
# boxplot ( t ( Q.stat ) ~ my.betas , pch = 20 , xlab = expression ( beta ), ylab = expression ( Q[ENV]) )
# abline ( h = tmp [[ 1 ]] [ 1, 5 ] )
# Q.stat <- do.call ( cbind , lapply ( tmp , function ( x ) x [ , 6 ] ) )
# 
# par ( mfrow = c ( 2, 1 ) )
# plot ( my.betas , power , xlab = expression ( beta ) , ylab = "Power"  , type = "l" , lwd = 2 )
# 
# boxplot ( t ( Q.stat ) ~ my.betas , pch = 20 , xlab = expression ( beta ), ylab = expression ( Q[ENV]) )
# abline ( h = tmp [[ 1 ]] [ 1, 5 ] )
```
<<<<<<< HEAD


Chunk for all of the teosintes, though currently not working as I need to make the matrices with population averages of genotypes in RRblup.  Initial population covariance matrices from Bayenv were made by Tim Beissenger, and the pop averages from my phenotypes stuff.  Bayenv uses different assumptions than rrBLUP, and therefore we want to stick to just rrBLUP.

=======


Chunk for all of the teosintes, though currently not working as I need to make the matrices with population averages of genotypes in RRblup.  Initial population covariance matrices from Bayenv were made by Tim Beissenger, and the pop averages from my phenotypes stuff.  Bayenv uses different assumptions than rrBLUP, and therefore we want to stick to just rrBLUP.

>>>>>>> 734b6aa2fbd6492078fbcbeff73f551250c1a4db
```{r}
# phenoteo <-read.csv("~/Documents/Projects/Genome_Size_Analysis/Github_ParallelGS/PhenotypeData/Teosinte_averages.csv",header=TRUE)
# 
# genoteo <- read.csv("~/Documents/Projects/Genome_Size_Analysis/Github_ParallelGS/SNP_data/TeosinteAll_Bayenv.csv",header=FALSE)
# 
# EnvVarTest(phenoteo$X180knobMB,genoteo,phenoteo$Altitude)
# EnvVarTest(phenoteo$X180knob.,genoteo,phenoteo$Altitude)
# EnvVarTest(phenoteo$TR1MB,genoteo,phenoteo$Altitude)
# EnvVarTest(phenoteo$TR1.,genoteo,phenoteo$Altitude)
# EnvVarTest(phenoteo$CentCMB,genoteo,phenoteo$Altitude)
# EnvVarTest(phenoteo$CentC.,genoteo,phenoteo$Altitude)
```

Chunk for just parv.

```{r}
# phenoparv <-read.csv("~/Documents/Projects/Genome_Size_Analysis/Github_ParallelGS/PhenotypeData/Teosinte_parv_avg.csv",header=TRUE)
# 
# genoparv <- read.csv("~/Documents/Projects/Genome_Size_Analysis/Github_ParallelGS/SNP_data/TeosinteParv_Bayenv.csv",header=FALSE)
# 
# EnvVarTest(phenoparv$X180knobMB,genoparv,phenoparv$Altitude)
# EnvVarTest(phenoparv$X180knob.,genoparv,phenoparv$Altitude)
# EnvVarTest(phenoparv$TR1MB,genoparv,phenoparv$Altitude)
# EnvVarTest(phenoparv$TR1.,genoparv,phenoparv$Altitude)
# EnvVarTest(phenoparv$CentCMB,genoparv,phenoparv$Altitude)
# EnvVarTest(phenoparv$CentC.,genoparv,phenoparv$Altitude)
```

Chunk for selection on mex.

```{r}
# phenomex <-read.csv("~/Documents/Projects/Genome_Size_Analysis/Github_ParallelGS/PhenotypeData/Teosinte_mex_avg.csv",header=TRUE)
# 
# genomex <- read.csv("~/Documents/Projects/Genome_Size_Analysis/Github_ParallelGS/SNP_data/TeosinteMex_Bayenv.csv",header=FALSE)
# 
# EnvVarTest(phenomex$X180knobMB,genomex,phenomex$Altitude)
# EnvVarTest(phenomex$X180knob.,genomex,phenomex$Altitude)
# EnvVarTest(phenomex$TR1MB,genomex,phenomex$Altitude)
# EnvVarTest(phenomex$TR1.,genomex,phenomex$Altitude)
# EnvVarTest(phenomex$CentCMB,genomex,phenomex$Altitude)
# EnvVarTest(phenomex$CentC.,genomex,phenomex$Altitude)
```

Power sims chunk.  The rough draft but working version of it, cleaning it up with a funciton in the next go around.

```{r}
#install.packages("MASS")
library(MASS)
# p = mvn(my + Selection * Altitude, Va * Covariance)
# p will be an output of our phenotypes
mu <- rep(0, 83) #I think since we want 1K sims, we want to make the length of this 1k 
Altitude <- phenoorder$Altitude
B <- 10
pizza <- mu + B * Altitude
#covariance matrix is given in A
VA=1 #additive genetics variance
simpheno <- mvrnorm(n=1000,pizza, VA*A) #n=number of samples we have, for maize 83, for allteo 16 pop
#pieces i am missing: where to stick in altitude?  I think it is at the mu

pvals=sapply(1:1000,function(X) EnvVarTest(simpheno[X,],A,Altitude)[3])
#EnvVarTest(simpheno[X,],A,Altitude,verbose=T)
#hist(pvals,breaks=100)
#plot(density(pvals))
```

Function for sims.  For the variables, here is an in text explanation:
Samples is how many reps we want to run
slope is our beta term, how strong selection is
add.gen.var we assume to be 1, but is Va
gen.covar.mat is our covariance matrix based on SNP genetypes, built earlier as matrix A
altitudes is a vector of altitudes

```{r}
library(MASS)
sim <- function ( samples , slope , add.gen.var , gen.covar.mat , altitudes , verbose=F) {
  mu <- rep(0, 83)
  Altitude <- altitudes
  GCM <- gen.covar.mat
  B <- slope
  env.vector <- mu + B * Altitude
  VA = add.gen.var
  simpheno <- mvrnorm(n=samples,env.vector, VA*GCM)
  pvals=sapply(1:samples,function(X) EnvVarTest(simpheno[X,],gen.covar.mat,Altitude)[3])
  allVa=sapply(1:samples,function(X) EnvVarTest(simpheno[X,],gen.covar.mat,Altitude)[4])
  hist(pvals,breaks=100, main="Distribution of P Values")
  signif <- length(subset(pvals, pvals < 0.05))
  avgVa <- mean(allVa)
  hist(allVa,breaks=100, main="Histogram of Va Values for 1000 Simulations")
  return ( c ( signif , avgVa ) )
}
```

Now that the simulations are working, I want to make a plot of beta/Va, which is the comparable value of s that we observe in our biologically real results.  To make the plot, I will run a few of the following sims with changing values for beta.  Store those values in vectors, and make a plot!  Code for simulations is commented out.

```{r}
#sim(1000,0,1,A,phenoorder$Altitude) # 53/1000
#sim(1000,0.0001,1,A,phenoorder$Altitude) # 78/1000
#sim(1000,0.0002,1,A,phenoorder$Altitude) # 153/1000
#sim(1000,0.0003,1,A,phenoorder$Altitude) # 291/1000
#sim(1000,0.0004,1,A,phenoorder$Altitude) # 485/1000
#sim(1000,0.0005,1,A,phenoorder$Altitude) # 657/1000
#sim(1000,0.0006,1,A,phenoorder$Altitude) # 828/1000
#sim(1000,0.0007,1,A,phenoorder$Altitude) # 918/1000
#sim(1000,0.0008,1,A,phenoorder$Altitude) # 973/1000
#sim(1000,0.0009,1,A,phenoorder$Altitude) # 990/1000
#sim(1000,0.001,1,A,phenoorder$Altitude) # 999/1000
```

Head to the grindstone on this, thinking about relationship between significance and Beta/Va.  

```{r}
accepted <- c(.053 , .078, .153, .291, .485, .657, .828, .918, .973, .990, .999)
beta <- c(0)
beta <- as.numeric(union(beta, seq(0.0001,0.001,0.0001)))
<<<<<<< HEAD
plot(accepted ~ beta, main="Proportion of Significant P-Values", xlab="B/SqRt(Va)", ylab="Proportion Significant",options(scipen=5),xlim=c(0,.001))
=======
plot(accepted ~ beta, main="Proportion of Significant P-Values under varying Beta", xlab="B/Va", ylab="Proportion Significant",options(scipen=5),xlim=c(0,.003))
>>>>>>> 734b6aa2fbd6492078fbcbeff73f551250c1a4db
lines(beta, accepted, lwd=1.5)
knobmb <- abs(-0.049495790/5431.778287163)
abline(v=knobmb, col="red",lwd=1.5)
knob. <- abs(-0.001419767/4.972794944)
abline(v=knob., col="red")
temb <- abs(-0.061480283/8727.985264731)
abline(v=temb, col="green",lwd=1.5)
te. <- abs(0.0004300344/2.2614178222)
abline(v=te., col="green")
centcmb <- abs(0.00008894284/5.18658746447)
abline(v=centcmb, col="peru",lwd=1.5)
centc. <- abs(0.00001546521/0.00568202942)
abline(v=centc., col="peru")
crmmb <- abs(0.0004516771/0.6025925993)
abline(v=crmmb, col="blue",lwd=1.5)
crm. <- abs(0.000002730368/0.000606835736)
abline(v=centc., col="blue")
<<<<<<< HEAD
betasmb <- c(0.049495790,0.061480283,0.00008894284)
vasmb <- c(5431.778287163,8727.985264731,5.18658746447)
fraction = betasmb/vasmb
cols <- c("dark blue","dark red","olivedrab")
barplot(fraction, col=cols, names.arg=c("180 Knob", "TE", "CentC"), ylab="B/Va")
betasall <- c(0.049495790,0.001419767,0.061480283,0.0004300344,0.00008894284,0.00001546521,0.0004516771,0.000002730368)
vasall <- c(5431.778287163,4.972794944,8727.985264731,2.2614178222,5.18658746447,0.00568202942,0.6025925993,0.000606835736)
=======
betas<- c(0.049495790,0.001419767,0.061480283,0.0004300344,0.00008894284,0.00001546521,0.0004516771,0.000002730368)
vas <- c(5431.778287163,4.972794944,8727.985264731,2.2614178222,5.18658746447,0.00568202942,0.6025925993,0.000606835736)
>>>>>>> 734b6aa2fbd6492078fbcbeff73f551250c1a4db
par(mfrow=c(3,1))
barplot(log10(betas),main="beta")
barplot(log10(vas),main="vas")
fraction <- betas/vas
barplot(log10(fraction),main="fraction")
par(mfrow=c(1,1))
plot(phenoorder$Altitude,phenoorder$X180knobMB)
abline(lm(phenoorder$X180knobMB~phenoorder$Altitude),col="red")
points(phenoorder$Altitude,phenoorder$CentCMB)
abline(lm(phenoorder$CentCMB~phenoorder$Altitude),col="blue")
lm(phenoorder$CentCMB~phenoorder$Altitude)
```

With these sims, we see that we can pick up beta/Va of pretty low, which will be a solid sentence in the paper.  I assume we have to redo these things for the teosinte matrices once I build them.
<<<<<<< HEAD

This next chunk will run all of the TE's through jeremy's code.  My goal with this will be to run the TE's through, get a count of how many of the tests are significant, and then plot a distribution of B/Va values.

```{r}
#to get the genome size info
setwd("~/Documents/Projects/Genome_Size_Analysis")
library(ggplot2)
#install.packages(gridextra)
dataall <- read.csv("Master_Data_noNA.csv")
dataal <- subset(dataall, dataall$X1C_GS!="NA")
data <- subset(dataal, dataal$X1C_GS<4) #changed to include the far outlier
dmays <- subset(data, data$Species=="mays")
#to get the FTE tags
#tmp <- read.csv("~/Documents/Projects/Genome_Size_Analysis/Github_ParallelGS/PhenotypeData/MergeFTE/Abund/289_FTE_aggregated.csv",header=TRUE,row.names="FTE_group")
#tmp2 = as.matrix(tmp)
#crap <- t(t(tmp2)/tmp2[1189,])
#perc_FTE <- as.data.frame(t(crap))
#write.csv(perc_FTE,"~/Documents/Projects/Genome_Size_Analysis/Github_ParallelGS/PhenotypeData/MergeFTE/Abund/289_percFTE_shiny.csv")
tmp <- read.csv("~/Documents/Projects/Genome_Size_Analysis/Github_ParallelGS/PhenotypeData/MergeFTE/Abund/289_percFTE_shiny.csv")

tmp2 <- merge(tmp,data, by="uniqid")
#write.csv(tmp2, "~/Documents/Projects/Genome_Size_Analysis/Github_ParallelGS/PhenotypeData/tmp_fte.csv")
#here, we just copy out the TE %'s
#crap <- read.csv("~/Documents/Projects/Genome_Size_Analysis/Github_ParallelGS/PhenotypeData/tmp_subsetfte.csv", row.names="uniqid")
#or
colnames(tmp2)
crap <- tmp2[,1:1189]
row.names(crap) <- crap$uniqid
crap
gsz <- tmp2$X1C_GS
crap2 <- crap * gsz
#write.csv(crap2,"~/Documents/Projects/Genome_Size_Analysis/Github_ParallelGS/PhenotypeData/tmp_fte_gz.csv")
#write.csv(data,"~/Documents/Projects/Genome_Size_Analysis/Github_ParallelGS/PhenotypeData/lecrap.csv")

gzfte <- merge(crap2,data, by.x="row.names",by.y="uniqid")
#write.csv(gzfte,"~/Documents/Projects/Genome_Size_Analysis/Github_ParallelGS/PhenotypeData/gzfte.csv")

#then go in and change the _ to . in the file.
#RIMMA0626.2 to RIMMA0626.1


#HERE WE START WITH FINALIZED MATRICES
gzfte <- read.csv("~/Documents/Projects/Genome_Size_Analysis/Github_ParallelGS/PhenotypeData/gzfte.csv")
library("rrBLUP")
geno <- read.csv("~/Documents/Projects/Genome_Size_Analysis/Github_ParallelGS/SNP_data/Landrace_noSWUS_matrix.csv",header=TRUE,row.names=1)
dt <-t(geno)
A <- A.mat(dt)


#to make sure order is the same
le1 <- as.data.frame(colnames(geno))
names(le1)[1] <- "names"
le2 <- as.data.frame(gzfte$Row.names)
le <- setdiff(le1,le2)

phenoorder <- merge(le1, gzfte, by.x="names", by.y="Row.names",sort=FALSE)

#colnames(phenoorder)# 5 to 1191

betas <- c()
va <- c()
pval <- c()
for(i in 5:1191){
  tmp = EnvVarTest(phenoorder[,i],A,phenoorder$Altitude)
  #if ( is.na ( tmp[3])) break
  betas <- c(betas , tmp[1])
  va <- c(va , tmp[4])
  pval <- c(pval , tmp[3])
}


a <- as.vector(which(pval<0.05))
b<- which(coefvar>1)
intersect(a,b)
a

colnames(phenoorder)[a]
colnames(phenoorder)[b]

for(i in 5:1191){
  coefvar <- c(coefvar,sd(phenoorder[,i])/mean(phenoorder[,i]))
}

plot(phenoorder$RLX_wugaab_AC187472.46, phenoorder)

range(phenoorder$RLX_wugaab_AC187472.46)

plot(coefvar)
length(subset(coefvar,coefvar>1))

par(mfrow=c(2,2))
tmp <- c()
nerms <- colnames(phenoorder)
which()
for(i in a){
  tmp <- nerms[i]
  print(tmp)
  plot(phenoorder$Altitude,phenoorder[,i],ylab=colnames(phenoorder)[i])
  abline(lm(phenoorder[,i]~phenoorder$Altitude ),col="red")
}



pval2 <- as.numeric(pval)
signf <- subset(pval2, pval2<0.05)
length(signf)

hist(betas/sqrt(va),breaks=50,col="gray",main="Distribution of Selection Coeffecients",xlab="Beta/SqRt(Va)")

signf <- subset(pval, pval<0.05)

EnvVarTest(phenoorder[,1111],A,phenoorder$Altitude)

EnvVarTest(phenoorder[,i],A,phenoorder$Altitude)
betas <- c(betas, EnvVarTest(phenoorder[,i],A,phenoorder$Altitude)[2])

EnvVarTest(phenoorder$TotallTeMB,A,phenoorder$Altitude)
EnvVarTest(phenoorder$X180knob.,A,phenoorder$Altitude)

phenoorder$Tot


#the NA's come because we have a TE that is present in the teo but not in the maize lines, check to see if i get all 0s in maize where we get the Nas

#include an error in measurement term, basically reducing our ability to say heritability is 1 cuz we might not be able to measure super well

#should be sqroot va (variance is a squared, beta is not, make it the same)
```



=======
>>>>>>> 734b6aa2fbd6492078fbcbeff73f551250c1a4db
